pkg/data/dashboard/repo.go

```diff
@@ -14,23 +15,36 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const defaultURL = "https://git.rancher.io/"
+
 var (
 	prefix = "rancher-"
 )
 
-func addRepo(wrangler *wrangler.Context, repoName, branchName string) error {
+func addRepo(wrangler *wrangler.Context, repoName, repoURL, branchName string) error {
+	if repoURL == "" || repoURL == defaultURL {
+		repoURL = defaultURL + strings.TrimPrefix(repoName, prefix)
+	} else {
+		logrus.Warnf(
+			"System charts URL set to %q, which is not the default (%q). "+
+				"If Rancher has issues finding charts, consider resetting this to the default value",
+			repoURL,
+			defaultURL,
+		)
+	}
+
 	repo, err := wrangler.Catalog.ClusterRepo().Get(repoName, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
 		_, err = wrangler.Catalog.ClusterRepo().Create(&v1.ClusterRepo{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: repoName,
 			},
 			Spec: v1.RepoSpec{
-				GitRepo:   "https://git.rancher.io/" + strings.TrimPrefix(repoName, prefix),
+				GitRepo:   repoURL,
 				GitBranch: branchName,
 			},
 		})
-	} else if err == nil && repo.Spec.GitBranch != branchName {
+	} else if err == nil && (repo.Spec.GitRepo != repoURL || repo.Spec.GitBranch != branchName) {
```

Comment:
We should also set right ?

repo.Spec.GitRepo = repoURL.

Reply:
Indeed we should! Fixed, thanks.